kolibri-electron/src/index.js

```diff
@@ -0,0 +1,90 @@
+const { app, BrowserWindow } = require('electron');
+const path = require('path');
+const child_process = require('child_process');
+const http = require('http');
+
+// Handle creating/removing shortcuts on Windows when installing/uninstalling.
+if (require('electron-squirrel-startup')) { // eslint-disable-line global-require
+  app.quit();
+}
+
+const KOLIBRI = 'http://localhost:5000';
+
+const waitForKolibriUp = (splash, mainWindow) => {
+  console.log("Pinging kolibri server");
+  http.get(`${KOLIBRI}/static/assets/logo.ico`, (response) => {
+    if (response.statusCode !== 200) {
+      console.log("Error: " + response.statusCode);
+      waitForKolibriUp(splash, mainWindow);
+    }
+
+    mainWindow.loadURL(KOLIBRI);
+
+  }).on("error", (error) => {
+    console.log("Error: " + error.message);
+    setTimeout(() => waitForKolibriUp(splash, mainWindow), 1000);
+  });
+}
+
+const createWindow = () => {
+  const splash = new BrowserWindow({
+    width: 800,
+    height: 400,
+    frame: false,
+    modal: true,
+    center: true,
+    alwaysOnTop: true,
+    skipTaskbar: true,
+  });
+
+  const mainWindow = new BrowserWindow({
+    width: 800,
+    height: 600,
+    autoHideMenuBar: true,
+    center: true,
+    show: false,
+    title: 'Endless Key',
+    icon: path.join(__dirname, 'kolibri-icon.png'),
+  });
+
+  mainWindow.once('ready-to-show', () => {
+    splash.destroy();
+    mainWindow.maximize();
+    mainWindow.show();
+    mainWindow.focus();
+  });
+
+  mainWindow.on('page-title-updated', (event) => {
+    event.preventDefault();
+  });
+
```

Comment:
One small detail here: external links are opening in the application. Instead, they should open in the user's default browser. For that, we need to use `mainWindow.webContents.setWindowOpenHandler`: https://www.electronjs.org/docs/api/web-contents#contentssetwindowopenhandlerhandler.

```suggestion
  const { shell } = require('electron');

  const windowOpenHandler = ({url}) => {
    if (url.startsWith(KOLIBRI)) {
      return {action: 'allow'};
    }
    shell.openExternal(url);
    return {action: 'deny'};
  };

  splash.webContents.setWindowOpenHandler(windowOpenHandler);
  mainWindow.webContents.setWindowOpenHandler(windowOpenHandler);
```

Reply:
Thanks a lot, I've modified a bit the suggested code to allow relative paths